src/components/ArticleMain.jsx

```diff
@@ -1,49 +1,66 @@
 import { useState, useEffect } from "react";
 import { fetchArticleById } from "../api.js";
 import { useParams } from "react-router-dom";
+import { Link } from "react-router-dom";
+import Loading from "./Loading.jsx";
+import PageNotFound from "./PageNotFound.jsx";
 
 const ArticleMain = () => {
-    const [article, setArticle] = useState({});
-    const { article_id } = useParams();
+  const [article, setArticle] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+  const { article_id } = useParams();
 
-    useEffect(() => {
-        fetchArticleById(article_id).then((data) => {
-            setArticle(data.article);
-        });
-    }
-    , [article_id]);
+  useEffect(() => {
+    fetchArticleById(article_id).then((data) => {
+      setArticle(data.article);
+      setIsLoading(false);
+    });
+  }, [article_id]);
 
+  if (isLoading) return <Loading />;
+  if (!article) return <PageNotFound />;
 
   const myDate = new Date(article.created_at);
   const day = myDate.getDate();
   const month = myDate.toLocaleString("default", { month: "long" });
   const year = myDate.getFullYear();
   const date = `${day}/${month}/${year}`;
   return (
-    <div style={{ margin: "auto", width: "50%" , paddingBottom: "10px"}}>
-    <div className="ui segment">
-      <h1 className="ui centered huge header">{article.title}</h1>
-      <div className="image centered">
-        <img className="ui centered huge image" src={article.article_img_url} />
-      </div>
-      <div className="content centered">
-      <i className="user circle outline icon"></i>
-        <a className="header">{article.author}</a>
-        <div className="meta">
-          <span className="date">Created in: {date}</span>
+    <div style={{ margin: "auto", width: "50%", paddingBottom: "10px" }}>
```

Comment:
See if you can update these divs to something more semantic

Reply:
I will try to do that more towards the end because I don't have enough knowledge for that right now, also I will try to keep changing them slowly as I need to use them for something else or when I have some extra time to research some alternatives.

Reply:
Ideally we should be doing this as we go, we don't expect you to have this knowledge automatically, we can always use previous repos, notes and google. w3schools has some very useful resources for this.